### PR TITLE
Github Verification Worker

### PR DESCRIPTION
**Pull request by kevinz917**
- CF Worker for Github verification through Gist. Messages are posted on Gist and retrieved using Github's Gist public API.

### CODE REVIEW

File: worker/src/lib.rs [hunk 1]
```diff
@@ -376,6 +377,79 @@ pub async fn dns_lookup(
...
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+
+        let dns_result = jserr!(github::retrieve_gist_message(gist_id.clone()).await);
+
+        let mut vc = jserr!(github::build_gist_vc(&pk, github_username.clone()));
+
+        // check for matching usernames
+        if github_username != dns_result.owner.login {
```
Comment (sbihel) [inline worker/src/lib.rs:399]: Usernames are case insensitive. And you can rename `dns_result`

File: worker/src/github.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+pub struct Owner {
+    pub login: String,
+}
+#[derive(Deserialize, Debug)]
+pub struct GistContent {
+    pub content: String,
+}
+
+pub fn extract_gist_message(gist_message: String) -> Result<String> {
+    let split = gist_message.split("=");
```
Comment (sbihel) [inline worker/src/github.rs:33]: The Gist can look more like a tweet, with the message embedded

File: worker/Cargo.toml [hunk 3]
```diff
@@ -52,4 +52,4 @@ opt-level = "z"
 
 [package.metadata.wasm-pack.profile.profiling]
 wasm-opt = ['-g', '-O']
-# opt-level = "z"
+# opt-level = "z"
```
Comment (sbihel) [inline worker/Cargo.toml:55]: Be careful of what you commit

File: worker/src/github.rs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                  }
+              }
+          }
+      ],
+      "issuanceDate": Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true),
+      "id": format!("urn:uuid:{}", Uuid::new_v4().to_string()),
+      "type": ["VerifiableCredential", "DnsVerification"],
+      "credentialSubject": {
+          "id": format!("did:pkh:tz:{}", &hash_public_key(pk)?),
+          "sameAs": format!("github:{}", github_username)
```
Comment (sbihel) [inline worker/src/github.rs:78]:
```suggestion
          "sameAs": format!("urn:github:{}", github_username)
```

File: worker/src/github.rs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+                      "gistId": "https://tzprofiles.com/gistId",
+                      "gistApiAddress": "https://tzprofiles.com/gistApiAddress",
+                      "gistMessage": "https://tzprofiles.com/gistMessage"
+                  }
+              }
+          }
+      ],
+      "issuanceDate": Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true),
+      "id": format!("urn:uuid:{}", Uuid::new_v4().to_string()),
+      "type": ["VerifiableCredential", "DnsVerification"],
```
Comment (sbihel) [inline worker/src/github.rs:75]:
```suggestion
      "type": ["VerifiableCredential", "GithubVerification"],
```

File: worker/src/lib.rs [hunk 6]
```diff
@@ -376,6 +376,95 @@ pub async fn dns_lookup(
...
+
+        evidence_map.insert(
+            "timestamp".to_string(),
+            serde_json::Value::String(Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true)),
+        );
+
+        evidence_map.insert(
+            "gistId".to_string(),
+            serde_json::Value::String(gist_id.to_string()),
+        );
```
Comment (sbihel) [inline worker/src/lib.rs:426]: There's the version id/sha in the response as well, under `history[-1].version`

**Review by sbihel — APPROVED** [review]

File: worker/src/lib.rs (hunk 6 above)
Comment (kevinz917) [reply worker/src/lib.rs:426]: Added to evidence map

File: worker/src/lib.rs [hunk 7]
```diff
@@ -442,6 +442,11 @@ pub async fn gist_lookup(
             ),
         );
 
+        evidence_map.insert(
+            "gistVersion".to_string(),
+            serde_json::Value::String(gist_result.history.last().unwrap().version.to_string()),
```
Comment (sbihel) [inline worker/src/lib.rs:447]:
```suggestion
            serde_json::Value::String(jserr!(gist_result.history.last()).version.to_string()),
```

**Pull request merged**